clang/test/CheckedC/inferred-bounds/basic.c

```diff
@@ -426,7 +426,7 @@ void f42(void) {
 // CHECK: | `-DeclRefExpr {{0x[0-9a-f]+}} '_Array_ptr<int>' lvalue Var {{0x[0-9a-f]+}} 'p' '_Array_ptr<int>'
 // CHECK: `-IntegerLiteral {{0x[0-9a-f]+}} 'int' 1
 
-  _Array_ptr<int> r : count(1) = &p[0];
+  _Array_ptr<int> r : count(1) = &p[0]; // expected-error {{it is not possible to prove that the inferred bounds of 'r' imply the declared bounds of 'r' after initialization}}
```

Comment:
The set `EquivExprs` contains `{ (_Array_ptr<int>)(&p[0]), r }`. Without the changes in this PR, there would be a warning: "cannot prove declared bounds for 'r' are valid after initialization" since the bases `p` and `r` of the inferred bounds `(p, p + 1)` and declared bounds `(r, r + 1)` are not equal.